runelite-client/src/main/java/net/runelite/client/plugins/huntertracker/HunterTrackerPlugin.java

```diff
@@ -0,0 +1,211 @@
+import java.time.temporal.ChronoUnit;
+import javax.imageio.ImageIO;
+import javax.swing.SwingUtilities;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.Client;
+import net.runelite.api.GameState;
+import net.runelite.api.VarClientInt;
+import net.runelite.api.VarPlayer;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.UsernameChanged;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.game.ItemManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.task.Schedule;
+import net.runelite.client.ui.NavigationButton;
+import net.runelite.client.ui.PluginToolbar;
+
+@PluginDescriptor(
+		name = "Hunter Tracker"
+)
+@Slf4j
+public class HunterTrackerPlugin extends Plugin
+{
+	@Inject
+	private PluginToolbar pluginToolbar;
+
+	@Inject
+	private ConfigManager configManager;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private HunterWorld hunterWorld;
+
+	@Inject
+	private ItemManager itemManager;
+
+	@Inject
+	private HunterTrackerConfig config;
+
+	private HunterTrackerPanel panel;
+
+	private NavigationButton navButton;
+
+	private WorldPoint lastTickLoc;
+
+	@Provides
+	HunterTrackerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(HunterTrackerConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		BufferedImage icon;
+		synchronized (ImageIO.class)
+		{
+			icon = ImageIO.read(getClass().getResourceAsStream("hunter.png"));
+		}
+
+		panel = new HunterTrackerPanel(client, itemManager, configManager, config, hunterWorld);
+
+		navButton = NavigationButton.builder()
+				.tooltip("Hunter Tracker")
+				.icon(icon)
+				.panel(panel)
+				.priority(4)
+				.build();
+
+		pluginToolbar.addNavigation(navButton);
+
+		updatePanel();
+	}
+
+	@Subscribe
+	public void onUsernameChanged(UsernameChanged e)
+	{
+		updatePanel();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		pluginToolbar.removeNavigation(navButton);
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick t)
+	{
+		if (client.getGameState() != GameState.LOGGED_IN)
+		{
+			lastTickLoc = null;
+			return;
+		}
+
+		// This VarClientInt is only -1 when you aren't actively in game (including at the post-login screen)
+		// The Birdhouse VarPs are only initialized once you've clicked CLICK HERE TO PLAY
+		if (client.getVar(VarClientInt.INVENTORY_TAB) == -1)
+		{
+			return;
+		}
+
+		WorldPoint loc = lastTickLoc;
+		lastTickLoc = client.getLocalPlayer().getWorldLocation();
+		if (loc == null || loc.getRegionID() != lastTickLoc.getRegionID())
+		{
+			return;
+		}
+
+		BirdhouseRegion region = hunterWorld.getRegions().get(loc.getRegionID());
+		if (region != null)
+		{
+			// Write config with new varp and timestamps
+			// huntertracker.<username>.<regionID>.<varpID>=<varpValue>:<startTime>:<updatedAt>
```

Comment:
You don't need to store updatedAt, just startTime. Remove it and all of the dead code related to it